### ISSUE TITLE
Add Fortran support

### ISSUE THREAD

**Issue by kynan** [conda-forge/fftw-feedstock#16]
Could fftw be configured without Fortran support or would this better be a separate feedstock?

**Comment by kynan** [comment]
@conda-forge/fftw Any thoughts on this?

**Comment by jschueller** [comment]
not easy on win32

**Comment by kynan** [comment]
fftw is not built for win32 according to `meta.yaml`. Are you planning on adding this?

**Comment by jakirkham** [comment]
There is discussion about adding Windows support, yes. However I disagree that this blocks adding Fortran support. Just because we add Fortran support for Unix OSes doesn't mean we need to extend that support to Windows. HDF5 is built with Fortran support. It is also built on Windows. However it doesn't support Fortran on Windows. IMHO it is ok if a similar situation occurs here.

If you are interested in making this happen @kynan , I'd take a close look at PR ( https://github.com/conda-forge/hdf5-feedstock/pull/28 ) and implement a similar PR here.

**Comment by leouieda** [comment]
I'd also be interested in this. It's a requirement for [SHTOOLS](https://github.com/SHTOOLS/SHTOOLS) and we'd love to have a conda package for that.

Is anyone tacking this at the moment? If not, I'd like to take a shot.

**Comment by xoviat** [comment]
Hey folks, we've been working hard to resolve the windows fortran nightmare. In a few hours, you'll be able to add `flang` to your build dependencies and compile normally, with no issues (in theory). Let me know if you have any questions.

**Comment by leouieda** [comment]
@xoviat thanks for the update and all the work! Please share some more info when you can.

**Comment by xoviat** [comment]
The project was literally just added. In ~15 minutes:

`conda install flang`

**Comment by xoviat** [comment]
(assuming conda-forge channel has been added).

**Comment by xoviat** [comment]
So there are a few issues:

1. `cmake` HEAD supports flang (just within the last few hours), but obviously not yet released.
2. What does "enabling fortran" actually do? Does it compile additional object files or add fortran files the include directory?

**Comment by jschueller** [comment]
2. I dont see additional libs, only headers

**Comment by xoviat** [comment]
@jschueller Can you `ls` the headers, and then we can just `cp` them?

**Comment by jakirkham** [comment]
Before we start using this `flang` compiler everywhere, I think we still need to resolve a larger discussion about how we can go about doing that without breaking compatibility with `defaults`. Given yours ( @xoviat's  ) and @isuruf's work on `flang`, it would be great to have you involved in this discussion and, if possible, come to the planned compiler meeting. Discussion and details can be found in issue ( https://github.com/conda-forge/conda-forge.github.io/issues/460 ). Other interested parties are of course welcome as well.

**Comment by xoviat** [comment]
@leouieda I have built shtools with flang. It will be Python 3 only of course on Windows.

https://ci.appveyor.com/project/MarkWieczorek/build-shtools/build/1.0.35

**Comment by egpbos** [comment]
It seems like the [current default Fortran compiler on Conda](https://github.com/conda/conda-docs/blob/master/docs/source/user-guide/tasks/build-packages/compiler-tools.rst#compiler-packages) is now gfortran on Linux and macOS, which can be installed by adding to meta.yaml:

```yaml
requirements:
  build:
    - {{ compiler('fortran') }}
```

Would it be possible to move forward in this direction for now? The discussion on `flang` could, of course, be continued in parallel. I can do a PR for adding (default) Fortran compiler support.

**Comment by egpbos** [comment]
The above reference from Amuse is the reason I need this now. There is no real rush, it's not a critical dependency, but since this is the last day of our sprint, it would be cool if we could have this merged today so all our tests will pass ;)

**Comment by njzjz** [comment]
Is there any progress in this issue? I need to build a Fortran program with fftw.

**Comment by egpbos** [comment]
I will make a new PR, the old one is outdated.

**Comment by egpbos** [comment]
New PR: #49

**Comment by egpbos** [comment]
Thanks for the quick merge @jschueller! Fortran support should now be available.
